### ISSUE TITLE
Sometimes it looks like there's too many batch commands generated per single draw call

### ISSUE THREAD

**Issue by ksuprynowicz** [overte-org/overte#1282]
For example single web entity here seems to use 19 command calls, including two setPipeline calls:
![image](https://github.com/user-attachments/assets/e9ab6190-4053-406b-958f-5892d95e3bef)

**Comment by HifiExperiments** [comment]
interesting!  web entity doRender is pretty simple, it accounts for the last setResourceTexture through draw at the end: https://github.com/overte-org/overte/blob/2986352e7ef68ac99d1a4f4da218e9b78be23ee1/libraries/entities-renderer/src/RenderableWebEntityItem.cpp#L322

I wonder if the setPipeline is coming from something else that early exits from drawing?

or it could be related to web entities being useOwnPipeline, maybe there’s some set up we do that isn’t necessary for those

**Comment by JulianGro** [comment]
Without having much idea about this stuff: Web entities have a refresh rate property. Maybe these commands appear multiple times, if the web entity refresh rate is set higher than the current render rate?

**Comment by HifiExperiments** [comment]
good idea but the refresh rate/actual drawing of the web page is handled separately by Qt, which then is converted to a texture by us somewhere different (actually using pure OpenGL calls, not batch commands, if I remember correctly).

**Comment by ksuprynowicz** [comment]
> good idea but the refresh rate/actual drawing of the web page is handled separately by Qt, which then is converted to a texture by us somewhere different (actually using pure OpenGL calls, not batch commands, if I remember correctly).

Can you point me to the place there these OpenGL calls are? This would be extremely helpful since I'm working on web textures for Vulkan branch currently. I'm doing it using shared GPU memory between OpenGL and Vulkan.

**Comment by HifiExperiments** [comment]
yes!  it's very hard to follow from the code but it ends up coming from `RenderEventHandler::qmlRender`.  the other functions in that class also have raw GL commands
